app/tests/api/auth/test_api_key_auth.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+from flask import g
+from werkzeug.exceptions import Unauthorized
+
+from api.auth.api_key_auth import API_AUTH_USER, api_key_auth
+
+
+def test_api_key_auth_success(app, api_auth_token):
+    # Passing it the configured auth token successfully returns a user
+    with app.app.app_context():  # So we can attach the user to the flask app
+        user_map = api_key_auth(api_auth_token, None)
+
+        assert user_map.get("uid") == API_AUTH_USER.user_id
+        assert g.get("current_user") == API_AUTH_USER
+
+
+def test_api_key_auth_invalid_token(api_auth_token):
+    # If you pass it the wrong token
+    with pytest.raises(
+        Unauthorized,
+        match="The server could not verify that you are authorized to access the URL requested",
+    ):
+        api_key_auth("not the right token", None)
+
+
+def test_api_key_auth_no_configuration():
```

Comment:
🤔 Hmm this test is failing for me:
> AssertionError: Regex pattern 'Authentication is not setup' does not match "401 Unauthorized: The server could not verify that you are authorized to access the URL requested. You either supplied the wrong credentials (e.g. a bad password), or your browser doesn't understand how to supply the credentials required.".

Reply:
Running anything inside the docker container uses your environment variables. So, inside always uses the `local.env` file. Need to make sure the env var is unset for this test to work.

Reply:
Yah I unset `API_AUTH_TOKEN` in `local.env` and it was still failing.

Reply:
Did you restart the entire container before running again? Env vars only get loaded at startup

Reply:
I'm pretty sure I ran `make clean-volumes init start` and then `make test`. I'll try it again

Reply:
Hmmmmmmmm it works now 🤷